### ISSUE TITLE
svelte-spa-router compatibility

### ISSUE THREAD

**Issue by MerlinB** [jakedeichert/svelvet#69]
See the issue described in https://github.com/ItalyPaleAle/svelte-spa-router/issues/92.

As suggested by ItalyPaleAle this problem could be related to dynamically loaded components.

[Here](https://github.com/MerlinB/svelte-spa-router-example) is a minimal example of svelvet 0.4.0 in combination with svelte-spa-router.

**Comment by jakedeichert** [comment]
Thanks for creating a minimal example! I have no idea what could be happening here, but i should find time to dig into it over the next few days 👍

**Comment by jakedeichert** [comment]
I just dug into this and it _looks_ like the problem is on svelte-spa-router's side.

If I change [these lines in Router.svelte](https://github.com/ItalyPaleAle/svelte-spa-router/blob/master/Router.svelte#L207-L211) to add `.component`:

```html
{#if componentParams}
  <svelte:component this="{component.component}" params="{componentParams}" />
{:else}
  <svelte:component this="{component.component}" />
{/if}
```

Then everything works!

<img width="743" alt="Screen Shot 2020-03-21 at 10 35 38 AM" src="https://user-images.githubusercontent.com/1631044/77228733-b753a980-6b5f-11ea-8fad-6e0e80edca92.png">


---

I followed svelte's [API docs](https://svelte.dev/docs#svelte_component) and you'll notice they also use `.component`:

<img width="1218" alt="Screen Shot 2020-03-21 at 10 36 04 AM" src="https://user-images.githubusercontent.com/1631044/77228750-d5b9a500-6b5f-11ea-956a-f8c59d6df55e.png">


I'm going to post a comment in https://github.com/ItalyPaleAle/svelte-spa-router/issues/92.

**Comment by jakedeichert** [comment]
**Update:** this solution would fix `npm run dev` mode.

`npm run build` actually has a separate issue. Snowpack (or rollup-plugin-svelte) is over-optimizing [`Router.svelte`](https://github.com/ItalyPaleAle/svelte-spa-router/blob/master/Router.svelte) and removing all the exported helpers 😱 

<img width="873" alt="Screen Shot 2020-03-21 at 10 57 25 AM" src="https://user-images.githubusercontent.com/1631044/77229197-cc7e0780-6b62-11ea-8bc2-27aff5cc37ac.png">

**Comment by jakedeichert** [comment]
Okay I figured out the problem with dev mode.

Svelvet compiles your src svelte files with [`dev: true`](https://github.com/jakedeichert/svelvet/blob/551a7bd1858989ca43aeb0f9473fa6574598515f/src/index.ts#L82). But your rollup config didn't have this option enabled, so `Router.svelte` was getting compiled in _production mode_. It turns out that dev mode and prod mode svelte components don't like each other 😅 

That's my bad as the [test/example doesn't have it](https://github.com/jakedeichert/svelvet/blob/551a7bd1858989ca43aeb0f9473fa6574598515f/tests/snapshot-snowpack-config/snowpack.config.js).

The fix for dev mode is this:

```js
const rollupPluginSvelte = require("rollup-plugin-svelte");

module.exports = {
  rollup: {
    plugins: [
      rollupPluginSvelte({
        dev: process.env.NODE_ENV !== "production"
      })
    ]
  }
};
```

I'll PR a fix to update the example/test.

Then I'll be able to report an issue to snowpack about the production issue.

**Comment by gawlk** [comment]
This still doesn't work in production mode right ?

**Comment by jakedeichert** [comment]
@gawlk right, AFAIK it's still not working. Someone just needs to report the issue to snowpack to see what they think... I have no idea what `Router.svelte` is being over-optimized right now :/

**Comment by gawlk** [comment]
Eh that's too bad. I would but since I'm so new to snowpack and all, I'm not sure I'm the right person for the job

**Comment by jakedeichert** [comment]
No worries! I or someone else will find time to do this :)

**Comment by gawlk** [comment]
What's weird is that when I copy the library in my `src` everything is fine 🤯

**Comment by jakedeichert** [comment]
Yeah, when copying the npm module to your src, it will work. The problem is that snowpack is over-optimizing that specific dependency. But snowpack doesn't run on your own `src/` code :)

**Comment by gawlk** [comment]
Oh right, makes sense, should've dig a little deeper before commenting here haha

**Comment by Manborough** [comment]
Hey everyone, i'm having this same issue in snowpack currently. Have any of you managed to get it working?

I've already set my snowpack config as such

```const rollupPluginSvelte = require("rollup-plugin-svelte")

module.exports = {
  "extends": "@snowpack/app-scripts-svelte",
  "scripts": {},
  "plugins": [],
  installOptions: {
    rollup: {
      plugins: [
      rollupPluginSvelte({
        dev: true
      })
    ]
    }
  }
}
```

**Comment by scallaway** [comment]
What's the status with this?

I see that due to Snowpack v2 being released this project is no longer going to be maintained, so is there a new solution/workaround to the problem above as I still seem to be running into it.

Happy to move this discussion to a different repo if that's more appropriate!

**Comment by FredKSchott** [comment]
Hey! Just digging into this myself, It looks like https://github.com/jakedeichert/svelvet/issues/69#issuecomment-602098919 is the fix. If you're using `@snowpack/plugin-svelte`, then we can make this fix directly in the Snowpack plugin on our end.

**Comment by scallaway** [comment]
Yep I think that's a good idea!
